# Clair3
## Description
This directory archives the pipelines we used to run `Clair`3 and benchmark the results to the gold standard truth.


## Execution environments
1. Minimap2
```bash
conda env create -f ./envs/minimap2.yaml;
```
2. Clair3
Prepare clair3 environment (docker image, for example) and run the image following https://github.com/HKU-BAL/Clair3?tab=readme-ov-file#option-2-singularity
Make sure you have already installed docker or singularity, and download the image
```bash
conda config --add channels defaults;
conda create -n singularity-env -c conda-forge singularity -y;
conda activate singularity-env;

# singularity pull docker pre-built image
singularity pull docker://hkubal/clair3:v1.0.4;

```
You also need to download several models manually because not all models were contained in the latest package(see https://github.com/HKU-BAL/Clair3?tab=readme-ov-file#ont-provided-models) or you can use the image we prepared at `../image/clair3.sif` .

3. hap.py
```bash
conda env create -f ./envs/hap.yaml;
```
## Clair3 models
According to the (Clair3 documentation)[https://github.com/HKU-BAL/Clair3], the correct Clair3 models for different flowcell and basecaller versions of HG002 data is listed below, the download link is also provided.

| Flowcell type | Basecaller | Basecalling mode | Correct Clair3 model |
|:---:|:---:|:---:|:-----:|
| R9 | Guppy2.3.7 | \ | [r941_prom_hac_g238](http://www.bio8.cs.hku.hk/clair3/clair3_models/) |
| R9 | Guppy4.2.2 | FAST | Not available |
| R9 | Guppy4.2.2 | HAC | [r941_prom_hac_g360+g422](http://www.bio8.cs.hku.hk/clair3/clair3_models/) |
| R9 | Guppy6.3.8 | FAST | Not available |
| R9 | Guppy6.3.8 | HAC | [r941_prom_sup_g5014](http://www.bio8.cs.hku.hk/clair3/clair3_models/) |
| R9 | Guppy6.3.8 | SUP | [r941_prom_sup_g5014](http://www.bio8.cs.hku.hk/clair3/clair3_models/) |
| R10 | Dorado0.4.3 | FAST | Not available |
| R10 | Dorado0.4.3 | HAC |[r1041_e82_400bps_hac_v410](https://github.com/nanoporetech/rerio/tree/master/clair3_models) |
| R10 | Dorado0.4.3 | SUP | [r1041_e82_400bps_sup_v410](https://github.com/nanoporetech/rerio/tree/master/clair3_models) |



## Pipelines
1. Download data and build directory structure
```bash
bash ./prepare/prepare.sh;
```

2. Run `Minimap2` alignmen
```bash
bash ./alignment/alignment.sh;
```

3. Run `Clair3` variant calling with different models
```bash
bash ./variant_calling/clair3.sh;
```

4. Run `hap.py` evaluation
```bash
bash ./benchmark/benchmark.sh;
```


## Results descriptions
For each directory in `../results/alignment`, sequences are aligned to hg38 reference genome.

For each directory in `../results/variant_calling`, variants are called by clair3 with different model pre-trained for different basecalling configuration.

For each directory in `../results/benchmark`, benchmark results are generated by hap.py, you can check `*.summary.csv` files to see F1 scores.



## Repeat our results
To repeat our results, please 
1. Install the aforementioned conda environment.

2. Run `run_all.sh`
```bash
bash ./run_all.sh;
```

